modules/apps/27-interchain-accounts/controller/keeper/handshake_test.go

```diff
@@ -471,3 +476,106 @@ func (suite *KeeperTestSuite) TestOnChanCloseConfirm() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestOnChanUpgradeInit() {
+	var (
+		path     *ibctesting.Path
+		metadata icatypes.Metadata
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			name: "failure: invalid metadata",
+			malleate: func() {
+				metadata.Address = TestOwnerAddress
+			},
+			expError: icatypes.ErrInvalidAccountAddress,
+		},
+		{
+			name: "failure: change connection id",
+			malleate: func() {
+				metadata.ControllerConnectionId = differentConnectionID
+			},
+			expError: connectiontypes.ErrInvalidConnection,
+		},
+		{
+			name: "failure: change host connection id",
+			malleate: func() {
+				metadata.HostConnectionId = differentConnectionID
+			},
+			expError: connectiontypes.ErrInvalidConnection,
+		},
+		{
+			name: "failure: invalid metadata string",
```

Comment:
Maybe this is more explicit:

```suggestion
			name: "failure: cannot decode version string",
```